front/app/containers/Admin/formBuilder/components/FormBuilderSettings/index.tsx

```diff
@@ -44,11 +44,38 @@ const FormBuilderSettings = ({
   isDeleteDisabled = false,
 }: Props) => {
   const locales = useAppConfigurationLocales();
+  const [currentTab, setCurrentTab] = useState<'content' | 'logic'>('content');
+  const { formatMessage } = useIntl();
+  const { watch } = useFormContext();
 
   if (isNilOrError(locales)) {
     return null;
   }
 
+  const getPageList = () => {
+    // TODO: Only select pages which come after the question. For this iteration though, we are not
+    // validating against cyclical form flows, so to not have an error state here,
+    // all pages should be available in the list.
+    const formCustomFields: IFlatCustomField[] = watch('customFields');
+    const pageArray: { value: string; label: string }[] = [];
+    formCustomFields?.map((field) => {
+      if (field.input_type === 'page') {
+        pageArray.push({
+          value: field.temp_id || field.id,
+          label: `${formatMessage(messages.page)} ${getIndexForTitle(
+            formCustomFields,
+            field
+          )}`,
+        });
+      }
+    });
+    pageArray.push({
+      value: 'survey_end',
```

Comment:
Minor, since this string is important in the sense that it always has to be this. Let's use a constant that is changed in one place if needed.